pkg/cmd/bazci/bazci.go

```diff
@@ -213,8 +214,9 @@ func (s *monitorBuildServer) handleBuildEvent(
 				if summary != nil && summary.ShardCount > 1 {
 					outputDir = filepath.Join(outputDir, fmt.Sprintf("shard_%d_of_%d", testResult.shard, summary.ShardCount))
 				}
-				if testResult.attempt > 1 {
-					outputDir = filepath.Join(outputDir, fmt.Sprintf("attempt_%d", testResult.attempt))
+				// Only upload results of last attempt.
+				if testResult.attempt != lastAttempt {
```

Comment:
This should in theory resolve the problem we saw earlier, e.g., https://github.com/cockroachdb/cockroach/issues/96142#issuecomment-1407541955